cartographer_ros/src/map_writer.cc

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "map_writer.h"
+
+#include <fstream>
+
+#include "glog/logging.h"
+#include "yaml-cpp/yaml.h"
+
+namespace cartographer_ros {
+
+namespace {
+
+void WriteOccupancyGridToPgm(const ::nav_msgs::OccupancyGrid& grid,
+                             const std::string& filename) {
+  LOG(INFO) << "Saving final map to '" << filename << "'...";
```

Comment:
This function does not know if this is the 'final map'. Hoist comment to the callsite?